add user authentication with Flask-Login and validations

9jana8 opened the pull request:
- Added password hashing with Bcrypt
- Introduced SQLAlchemy db instance and User model
- Integrated Flask-Login: login_user, logout_user, current_user, login_required
- Added custom RegistrationForm validations (unique username and email checks)
- Enabled user login and session handling
- Restricted navigation options: show logout/account when logged in, login/register otherwise

9jana8 committed:
add user authentication with Flask-Login and validations

- Added password hashing with Bcrypt
- Introduced SQLAlchemy db instance and User model
- Integrated Flask-Login: login_user, logout_user, current_user, login_required
- Added custom RegistrationForm validations (unique username and email checks)
- Enabled user login and session handling
- Restricted navigation options: show logout/account when logged in, login/register otherwise

urosisakovic commented:
After this PR, rename `flask` branch to `master`

urosisakovic commented:
Use `jana/<feature name>` for a new branch

urosisakovic commented:
Set `master`  branch to require PR in order to merge

urosisakovic commented on instance/site.db:
This file should be .gitignored

urosisakovic replied:
I suggest creating `tools/initalize_db.py` script, that deletes and existing `.db` file an creates a new

urosisakovic commented:
Add a README.md with instruction on how to run. Use help from ChatGPT

urosisakovic commented:
Sidebar looking weird
<img width="1321" height="1069" alt="image" src="https://github.com/user-attachments/assets/d1fe7ad9-3cae-497d-82a7-fc374c11f225" />

urosisakovic commented:
Add .gitignore

urosisakovic commented:
Add one issue for each of the work items that you have next.

Each of the following PRs should be closing exactly one issue.

urosisakovic commented on flaskapp/forms.py:
```diff
@@ -9,6 +10,16 @@ class RegistrationForm(FlaskForm):
     confirm_password = PasswordField('Confirm Password', validators=[DataRequired(), EqualTo('password')])
     submit = SubmitField('Sign Up')
 
+    def validate_username(self, username):
```
Add type hints

urosisakovic replied:
Won't be spamming, add them everywhere

urosisakovic commented on flaskapp/forms.py:
```diff
@@ -9,6 +10,16 @@ class RegistrationForm(FlaskForm):
+    def validate_username(self, username):
+        user_exists = User.query.filter_by(username=username.data).first()
+        if user_exists:
+            raise ValidationError('This username is taken. Please choose unique one.')
```
```suggestion
            raise ValidationError('This username is taken. Please choose another one.')
```

urosisakovic commented on flaskapp/models/__init__.py:
```diff
@@ -0,0 +1,2 @@
+from .user import User
+from .note import Note
```
Add new line at the end of the file

urosisakovic replied:
Won't be spamming, do it everywhere

urosisakovic commented on flaskapp/routes.py:
```diff
@@ -26,19 +28,39 @@ def about():
     if form.validate_on_submit():
-        if form.email.data == '9jana8@gmail.com' and form.password.data == 'password':
+        user = User.query.filter_by(email=form.email.data).first()
+        if user and bcrypt.check_password_hash(user.password, form.password.data):
```
:+1:

urosisakovic reviewed (changes requested):
Excellent first PR! :heart: 

There are a lot of setup things to be done, but once you get those out of the way, the next PRs will be much smoother :)

9jana8 committed:
roject improvements and workflow setup

- added .gitignore
- implemented 'create a note' feature for logged-in users; replaced dummy post data with DB-backed notes
- added type hints throughout the code
- fixed sidebar display on /login route
- introduced 'issue' logic to project workflow on GitHub
- enforced master branch to require PRs
- renamed 'flask' branch to 'main'

9jana8 committed:
- added README.md
- added requirements.txt

9jana8 committed:
"feat(post): implement CRUD functionality (closes #7)"
